### PR TITLE
Fix FP env issues

### LINKED ISSUE #11086: Handling of floating point environment
Most processors have a floating-point "environment" which includes the rounding mode, exception state, and probably other stuff.
C has a bunch of floating point environment-manipulation functions which expose that environment. LLVM has a bunch of pragmas, attributes, flags, and intrinsics which control what can be assumed about that environment, what optimizations are allowed, what can happen when various FP operations execute, and what instructions are selected.
Wasm actually does not have such an environment; instead the rounding mode and exception behavior is fixed for all instructions (although there are a few instructions which have both trapping and nontrapping variants).
Emscripten has dummy implementations of [many](https://github.com/emscripten-core/emscripten/blob/master/system/lib/libc/musl/src/fenv/fenv.c) of those C functions, which do nothing because there is no environment to expose. Other than that, we don't do anything in particular in LLVM or clang to support or not support the various controls on modes or operations.

In general, we should
1) not crash when these various controls are used
2) correctly implement whatever controls can be implemented correctly
3) do something reasonable for controls that can't be implemented (an interesting question is whether to ignore them or throw some sort of error)
4) Take advantage of whatever properties wasm has to apply whatever optimizations are are possible.

In particular we are currently violating 1) (e.g. the `-frounding-math` causes LLVM's [constrained FP intrinsics](https://llvm.org/docs/LangRef.html#constrained-floating-point-intrinsics) to be used, and we currently fail to lower the strict SETCC node during ISel dag legalization. The thing that caused me to write this up today was that upstream clang recently started supporting `#pragma STDC FENV_ACCESS ON` which is a per-function way to enable the same behavior, and that pragma is actually used by musl and causes build failures.
So in the very near term we need to at last either fix that ISel or comment those pragmas out of musl. Longer term we should figure out all the ways this functionality is exposed and what it should mean on wasm.

### PR DESCRIPTION
**Pull request by kripken**
1. Define only floating point exceptions we support (none). That
  avoids a lot of float pragmas we don't support.
2. Ifdef out the remaining such pragmas.

Helps #11086 , but does not fix it as user code may still be broken
due to LLVM failing ISel on those pragmas now.

### CODE REVIEW

**Comment by kripken** [comment]
This makes libc 164 bytes smaller.

**Comment by dschuff** [comment]
I think some change to the fenv defs might make sense (although, should we do something similar for rounding modes?), but I'm not so thrilled about ifdef-ing out the pragmas for the long term. I think we will still want to handle the constrained intrinsics in one form or another. Perhaps with an ISel fix. 
This PR doesn't fix the fact that users might have code that has that pragma too, where we previously ignored it but now crash. So at least we should not mark #11086 as fixed.
But this should work to unblock the LLVM roll for now. let me test it with the new LLVM.

File: system/lib/libc/musl/arch/emscripten/bits/fenv.h [hunk 1]
```diff
@@ -1,11 +1,4 @@
-#define FE_INVALID    1
-#define __FE_DENORM   2
-#define FE_DIVBYZERO  4
-#define FE_OVERFLOW   8
-#define FE_UNDERFLOW  16
-#define FE_INEXACT    32
```
Comment (sbc100) [inline system/lib/libc/musl/arch/emscripten/bits/fenv.h:6]: Are these designed to be disabled by simply not being defined?  That seems unusual, but I don't know this part of the standard.

Comment (kripken) [reply system/lib/libc/musl/arch/emscripten/bits/fenv.h:6]: Yeah, I learned this today as well. It is quite unusual for the C spec, I agree.

>  Libraries may define in <fenv.h> only the macro values above they support (the others may not be defined).

http://www.cplusplus.com/reference/cfenv/FE_INEXACT/

And musl in fact checks whether things are defined in order to ifdef relevant code. Which actually is helpful in this case!

Comment (dschuff) [reply system/lib/libc/musl/arch/emscripten/bits/fenv.h:6]: Not sure if they are designed that way exactly, but  musl code seems to use them that way: see e.g. https://github.com/emscripten-core/emscripten/blob/master/system/lib/libc/musl/src/fenv/__flt_rounds.c and in particular https://github.com/emscripten-core/emscripten/blob/2bca083cbbd5a4133db61fbd74d04f7feecfa907/system/lib/libc/musl/src/math/lrint.c gates its use of the FENV_ACCESS pragma on `FE_INEXACT` being defined.

**Pull request merged**